### PR TITLE
Create rest api for crud favourites

### LINKED ISSUE #62: Integrate Cockroach Db with the dashboard page
The cockroach db can be used to store the data about people's favourite repos/issues

### LINKED ISSUE #79: Add a new repositories page and favourites page
Add a new page where users can get a list of repositories that they have contributed to and favourite some repositories too.

### PR DESCRIPTION
**Pull request by Green-Ranger11**
### Why This PR Adds Value

This PR creates a api with .Net which is used for adding, deleting and getting favourites for a user.

### What This PR Adds

- Get a user's favourites
- Add a favourite
- Remove a favourite

### Screenshot

image

Navigate to (url:port)/swagger/index.html to see more in-depth details on the different endpoints and their expected requests and responses.

### How This PR Could Be Improved

- Move into another repository
- EF Core doesn't have full support for cockroachdb

### CODE REVIEW

**Comment by anandrajaram21** [comment]
@Green-Ranger11, shouldn't this PR be into staging?

**Comment by Green-Ranger11** [comment]
> @Green-Ranger11, shouldn't this PR be into staging?

Have fixed this

File: README.md [hunk 1]
```diff
@@ -45,6 +45,25 @@ Then, run the development server:
...
+
+Then, run the development server:
+```bash
+cd server/api
+dotnet run
+```
+
 ### GitHub Authentication
 
 To get the GitHub authentication to work, you need to create a `.env` file in the `gitdash` directory with the following content
```
Comment (anandrajaram21) [inline README.md]: Can you rename `gitdash` here to `client`?

File: server/.vscode/launch.json [hunk 2]
```diff
@@ -0,0 +1,43 @@
+{
```
Comment (anandrajaram21) [inline server/.vscode/launch.json]: Is this file created by default while setting up the boilerplate .NET API? If not, can you please remove the `.vscode` folder and add `.vscode` to the gitignore?

File: server/.vscode/settings.json [hunk 3]
```diff
@@ -0,0 +1,9 @@
+{
```
Comment (anandrajaram21) [inline server/.vscode/settings.json]: Is this file created by default while setting up the boilerplate .NET API? If not, can you please remove the .vscode folder and add .vscode to the gitignore?

File: server/.vscode/tasks.json [hunk 4]
```diff
@@ -0,0 +1,42 @@
+{
```
Comment (anandrajaram21) [inline server/.vscode/tasks.json]: Is this file created by default while setting up the boilerplate .NET API? If not, can you please remove the .vscode folder and add .vscode to the gitignore?

**Review by anandrajaram21 — CHANGES_REQUESTED** [review]
I have just blindly approved the .NET API changes as I don't have any experience with .NET. Sorry for that 😅. I have requested some changes. Can you please fix them and request a review again? Great work so far!

File: README.md (hunk 1 above)
Comment (Green-Ranger11) [reply README.md]: Done. In my latest commit

Comment (anandrajaram21) [reply README.md]: But the file review section still shows `gitdash` and not `client` in the README.md

File: server/.vscode/launch.json (hunk 2 above)
Comment (Green-Ranger11) [reply server/.vscode/launch.json]: Yep thats my bad, I've added it now

File: server/.vscode/settings.json (hunk 3 above)
Comment (Green-Ranger11) [reply server/.vscode/settings.json]: Removed it from git

File: server/.vscode/tasks.json (hunk 4 above)
Comment (Green-Ranger11) [reply server/.vscode/tasks.json]: Removed from git

**Review by anandrajaram21 — COMMENTED** [review]
These changes look really good 👍

File: README.md (hunk 1 above)
Comment (anandrajaram21) [reply README.md]: Can you change `gitdash` on line 69 to `client`?

Comment (Green-Ranger11) [reply README.md]: Oh my bad, I thought you were referring the folder. Fixed it now

**Review by anandrajaram21 — APPROVED** [review]
These changes are now perfect. Amazing job @Green-Ranger11 on the API.

**Pull request merged**